panels/base_panel.py

```diff
@@ -117,6 +121,8 @@ def reload_icons(self):
             width = pixbuf.get_width()
             height = pixbuf.get_height()
             button.set_image(self._gtk.Image(name, width, height))
+        
+        self.battery_percentage()
```

Comment:
Would there be a better location to place this? or is it okay to leave as long as the config check is added?

The reason it was placed there is when the theme was changed the icon did not update if it was placed inside of ```def __init__```. Since that is not reinitialized after changing the theme as far as i can tell, so the next time the icon would update would be the next time the timer calls ```self.battery_percentage()```.

Reply:
it should be called just before the timeout, think of it as: do it now, then every x time units

Reply:
Gotcha, yeah in the commits I pushed it now creates/inits the gtk box and labels in '''def __init__'''. Then in '''def reload_icons''' just calling '''battery_percentage()''' which just updates.